### PR TITLE
Use AutoClosable for to ReactorSender, ReactorReceiver and ServiceBusSessionReceiver

### PR DESCRIPTION
**Pull request by YijunXieMS**
Also allows retry when building network connection timeout

### CODE REVIEW

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java [hunk 1]
```diff
@@ -62,7 +63,7 @@
 /**
  * Handles scheduling and transmitting events through proton-j to Event Hubs service.
  */
-class ReactorSender implements AmqpSendLink, AsyncAutoCloseable {
+class ReactorSender implements AmqpSendLink, AsyncAutoCloseable, Closeable {
```
Comment (conniey) [inline sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java:66]: I don't remember was it AutoCloseable or Closeable?

**Review by conniey — APPROVED** [review]

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java (hunk 1 above)
Comment (srnagar) [reply sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java:66]: this should be AutoCloseable

**Review by srnagar — APPROVED** [review]

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java [hunk 2]
```diff
@@ -108,7 +108,7 @@ public ReactorConnection(String connectionId, ConnectionOptions connectionOption
                 final Mono<AmqpEndpointState> activeEndpoint = getEndpointStates()
                     .filter(state -> state == AmqpEndpointState.ACTIVE)
                     .next()
-                    .timeout(operationTimeout, Mono.error(new AmqpException(false, String.format(
+                    .timeout(operationTimeout, Mono.error(new AmqpException(true, String.format(
```
Comment (hemanttanwar) [inline sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java:111]: Have we tested this where we are converting this exception into transient error ?

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java (hunk 1 above)
Comment (hemanttanwar) [reply sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java:66]: Other places we are using `AutoCloseable` but here we are using  `Closeable`

**Review by hemanttanwar — APPROVED** [review]

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java (hunk 2 above)
Comment (YijunXieMS) [reply sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java:111]: Yes I tested locally. When my home computer has wifi but the router actually is disconnected from internet, this timeout may kick in and there is no retry. After this change there is retry.

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java (hunk 1 above)
Comment (YijunXieMS) [reply sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java]: My bad. Corrected to AutoClosable

**Pull request merged**